tools/validators/instance_validator/validate/instance_parser.py

```diff
@@ -250,13 +250,17 @@ def GetConfigMode(self) -> ConfigMode:
                                'Finalize()')
     return self._config_mode
 
-  def AddFile(self, filename: str) -> None:
+  def AddFile(self, filenumber: int, filename: str) -> None:
     """Add a new file for parsing by the state machine.
 
     Args:
       filename: a path to a file for reading with open()
     """
-    entity_instance_block = ''
+    entity_instance_block = {
+      'source_filenumber': filenumber,
+      'source_filename': filename,
+      'block': '',
+    }
```

Comment:
since this is shared with multiple methods, it should be defined as a class (if it was immutable, a named tuple would have been OK too).  An addLine() and isEmpty() method would improve the readability of the original code too:

block = ConfigBlock(filename, filenumber)

block.AddLine(line)

if(block.isEmpty()):
  do something...

Reply:
I didn't originally implement it as a class because that seemed like overkill to me. However, your points are valid and making it a class will help "future proof" it incase we need more attributes in the future. 

Regarding method names, are you sure you want to use `camelCase`? Pythonic style guides favor `wide_text` for most general variable/function/method names. (e.g. https://google.github.io/styleguide/pyguide.html)